core/templates/domain/story/editable-story-backend-api.service.ts

```diff
@@ -16,203 +16,246 @@
+        story_id: storyId
       });
+
+    const putData = {
+      version: storyVersion,
+      commit_message: commitMessage,
+      change_dicts: changeList
     };
 
-    var _changeStoryPublicationStatus = function(
-        storyId, newStoryStatusIsPublic, successCallback, errorCallback) {
-      var storyPublishUrl = UrlInterpolationService.interpolateUrl(
-        STORY_PUBLISH_URL_TEMPLATE, {
-          story_id: storyId
-        });
+    this.http.put<UpdateStoryBackendResponse>(
+      editableStoryDataUrl, putData).toPromise().then(
+      response => successCallback(response.story),
+      errorResponse => errorCallback(errorResponse.error.error)
+    );
+  }
 
-      var putData = {
-        new_story_status_is_public: newStoryStatusIsPublic
-      };
-      $http.put(storyPublishUrl, putData).then(function(response) {
-        if (successCallback) {
-          successCallback();
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+  private _changeStoryPublicationStatus(
+      storyId: string,
+      newStoryStatusIsPublic: boolean,
+      successCallback: (value: number) =>void,
+      errorCallback: (reason: string) => void): void {
+    const storyPublishUrl = this.urlInterpolationService.interpolateUrl(
+      StoryDomainConstants.STORY_PUBLISH_URL_TEMPLATE, {
+        story_id: storyId
       });
+    const putData = {
+      new_story_status_is_public: newStoryStatusIsPublic
     };
+    // eslint-disable-next-line max-len
+    this.http.put(storyPublishUrl, putData, { observe: 'response'}).toPromise().then(
+      response => successCallback(response.status),
+      errorResponse => errorCallback(errorResponse.error.error));
+  }
 
-    var _validateExplorations = function(
-        storyId, expIds, successCallback, errorCallback) {
-      var validateExplorationsUrl = UrlInterpolationService.interpolateUrl(
-        VALIDATE_EXPLORATIONS_URL_TEMPLATE, {
+  private _validateExplorations(
+      storyId: string,
+      expIds: string[],
+      successCallback: (value: number) => void,
+      errorCallback: (reason: string) => void): void {
+    const validateExplorationsUrl =
+      this.urlInterpolationService.interpolateUrl(
+        StoryDomainConstants.VALIDATE_EXPLORATIONS_URL_TEMPLATE, {
           story_id: storyId
         });
 
-      $http.get(validateExplorationsUrl, {
+    this.http.get<void>(
+      validateExplorationsUrl, {
+        observe: 'response',
         params: {
           comma_separated_exp_ids: expIds.join(',')
         }
-      }).then(function(response) {
-        if (successCallback) {
-          successCallback(response.data.validation_error_messages);
-        }
-      }, function(errorResponse) {
-        if (errorCallback) {
-          errorCallback(errorResponse.data);
-        }
+      }
+    ).toPromise().then(
+      response => successCallback(response.status),
```

Comment:
Earlier `successCallback` was called with `validation_error_messages`. Why has that changed?

Reply:
corrected!